libsel4arm-vmm/src/arch/arm/devices/vgic.c

```diff
@@ -292,21 +297,11 @@ static int vgic_vcpu_inject_irq(struct device *d, vm_t *vm, struct virq_handle *
         return err;
     } else {
         /* Add to overflow list */
-        struct lr_of **lrof_ptr;
-        struct lr_of *lrof;
-        lrof_ptr = &vgic->lr_overflow;
-        while (*lrof_ptr != NULL) {
-            lrof_ptr = &(*lrof_ptr)->next;
-        }
-        list_size++;
-        lrof = (struct lr_of *)malloc(sizeof(*lrof));
-        assert(lrof);
-        if (lrof == NULL) {
-            return -1;
-        }
-        lrof->virq_data = irq;
-        lrof->next = NULL;
-        *lrof_ptr = lrof;
+        ZF_LOGF_IF(vgic->lr_overflow.head == vgic->lr_overflow.tail, "No overflow irqs to service");
```

Comment:
Slightly confused about this check. Won't this condition be hit the first time we overflow? (from the call path `vm_inject_IRQ` -> `vgic_dist_set_pending_irq` -> `vgic_vcpu_inject_irq`)

Reply:
yup that makes no sense, will remove